lib/member_name_parts.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+require 'scraped'
+
+class MemberNameParts < Scraped::HTML
+  field :prefix do
+    partitioned.first
+  end
+
+  field :name do
+    # .sub called for specific case where Dr. is not separated from name by space
+    partitioned.last.sub('Dr.Bharrat', 'Bharrat')
+  end
+
+  field :gender do
+    prefixes_chomped.map { |p| gender_map[p] }.compact.first
+  end
+
+  private
+
+  def prefixes
+    %w(Assoc Prof Professor Rev Bishop Prince Dr Lt Col Colonel (Ret’)).to_set
+                                                                       .merge(female_prefixes)
+                                                                       .merge(male_prefixes)
+  end
```

Comment:
This is operating at several different levels of specificity at the same time. It would be better to split this so there are three different sets of prefixes which then get joined in one place.